java/com/google/flatbuffers/Table.java

```diff
@@ -239,6 +243,28 @@ protected static int compareStrings(int offset_1, int offset_2, ByteBuffer bb) {
     if (len_1 > len_2) return 1;
     return 0;
   }
+  
+  /**
+   * Compare string from the buffer with the 'String' object.
+   *
+   * @param offset_1 An 'int' index of the first string into the bb.
+   * @param key Second string.
+   * @param bb A {@code ByteBuffer} to get the first string.
+   */
+  protected static int compareStrings(int offset_1, String key, ByteBuffer bb) {
+    offset_1 += bb.getInt(offset_1);
+    int len_1 = bb.getInt(offset_1);
+    int len_2 = key.length();
+    int startPos_1 = offset_1 + Constants.SIZEOF_INT;
+    int len = Math.min(len_1, len_2);
+    for (int i = 0; i < len; i++) {
+      if (bb.array()[i + startPos_1] != key.charAt(i))
```

Comment:
the performance of this function is entirely down to this loop, so it may be worth caching `bb.array()` outside of the loop just in case the compiler is too dumb to do it.

Also, this code does not support UTF-8 or unicode at all. `bb.array()` may contain unicode sequences, which will typically cause the comparison to fail, or worse yet, it could actually cause a false positive.

To do this correctly, we'd either have to do UTF-8 parsing on the bytes as we compare them (probably fastest since we don't have to unpack all characters that way, but I'd hate to add yet another UTF-8 parser to the code). Alternatively, we unpack the entire string into a `String` first. This is easier, but is very inefficient, and will cause large amounts of garbage generated by a binary search.

The third alternative is to leave it as-is and document that this lookup only works with ASCII subset. I doubt that will be acceptable.

Actually, a 4th option is to first convert the search string (`key`) into a UTF-8 byte array, and then do bytewise comparison. This is decently efficient, since it only needs to be done once for the entire search. This sounds the best to me. You can early out if lengths are different, and maybe use Arrays.equals for the actual comparison.